### DISCUSSION TITLE
Getting cypress to use GPU on Linux

### DISCUSSION THREAD

**Discussion by paul-carlton** [cypress-io/cypress#29888] [Questions and Help]
I am trying to run cypress on AWS using Amazon Linux 2 AMI with NVIDIA TESLA GPU Driver  but I am not seeing any GPU usage

**Comment by portyu9** [comment]
Hi @paul-carlton 

There is a Cypress-specific reason you are not seeing GPU use: Cypress currently adds:

```text
--disable-gpu
```

when launching Chromium-family browsers on Linux.

You can experimentally remove that argument with `before:browser:launch`:

```ts
import { defineConfig } from 'cypress'

export default defineConfig({
  e2e: {
    setupNodeEvents(on) {
      on('before:browser:launch', (browser, launchOptions) => {
        if (
          browser.family === 'chromium' &&
          browser.name !== 'electron'
        ) {
          launchOptions.args = launchOptions.args.filter(
            (arg) => arg !== '--disable-gpu'
          )
        }

        return launchOptions
      })
    },
  },
})
```

This only **allows** the browser to use the GPU; it does not guarantee that hardware acceleration will actually be available.

You still need the OS/container/browser stack to expose the NVIDIA GPU correctly. After removing the flag, verify what Chromium actually sees rather than relying only on host-level GPU utilization.

For example, check the browser's GPU/WebGL information and confirm that the renderer is NVIDIA rather than a software implementation such as SwiftShader/llvmpipe.

There is also an important current-version difference for your environment: **Cypress 14+ prebuilt Linux binaries do not support Amazon Linux 2**, because Amazon Linux 2 uses a glibc version below the current minimum. For a current Cypress installation, move the run to a supported Linux distribution/container first.

So:

1. on the historical environment, remove `--disable-gpu` and verify Chromium's renderer;
2. make sure the NVIDIA device/driver is actually available to the browser process;
3. for current Cypress versions, migrate off Amazon Linux 2 to a supported Linux image.

No screenshots or attachments were included in this discussion.

Please mark this answer as accepted if it helped, thank you.

**Reply by paul-carlton** [reply]
Thanks so much, I no longer work for the company I was doing this for but this useful information and if I get the opportunity I will try out this
